packages/multi-test/src/wasm.rs

```diff
@@ -425,14 +425,17 @@ where
         }
 
         // recurse in all messages
-        for resend in response.messages {
-            let subres = self.execute_submsg(router, storage, block, contract.clone(), resend)?;
-            events.extend_from_slice(&subres.events);
-        }
-        Ok(AppResponse {
-            events,
-            data: response.data,
-        })
+        let data = response
+            .messages
+            .into_iter()
+            .try_fold(response.data, |data, resend| {
```

Comment:
Interesting rework of loop to functional, but looks good

Reply:
I could do that or introduce mutable variable which is updated. I find folding way more readable that mutating, as if you understand folding, nothing surprises you. Analyze mutation always requires you to carefully read the loop. Otherwise I won't bother changing it.